include/NovelRT/Ecs/SystemScheduler.h

```diff
@@ -18,12 +18,6 @@ namespace NovelRT::Ecs
     class SystemScheduler
     {
     private:
-        struct QueueLockPair
-        {
-            std::optional<Atom> systemUpdateId;
-            std::mutex threadLock;
```

Comment:
No more locking required?

Reply:
Yesn't.

We no longer lock in the traditional sense, but atomic operations might perform cheaper atomic locks on the XOR operations, or may just be volatile writes.

Reply:
So are we hoping that we hoping that we never lock in this case (lock-free)? Or do we want to just defer to the CPU to determine this at runtime?

Reply:
It's a case of the atomic lock would be so cheap on the XOR operations that it literally doesn't matter and you may as well consider this lockless due to the fact were using no mutexes at this point.